src/main/java/seedu/duke/Internship.java

```diff
@@ -0,0 +1,129 @@
+package seedu.duke;
+
+import seedu.exceptions.InvalidStatus;
+
+import java.time.YearMonth;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Class to store the relevant information for an internship.
+ */
+//@@author jadenlimjc
+public class Internship {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MM/yy");
+
+    private int id = -1;
+    private String role;
+    private String company;
+    private YearMonth startDate;
+    private YearMonth endDate;
+    private String skills;
+    private String status;
+
+
+    /**
+     * Constructs an <code>Internship</code> instance with information provided.
+     * Skills and Status fields are initialised with default values.
+     *
+     * @param role    description of role.
+     * @param company company name.
+     * @param start   start year/month.
+     * @param end     end year/month.
+     */
+    public Internship(String role, String company, String start, String end) {
+        this.role = role;
+        this.company = company;
+        setStartDate(start);
+        setEndDate(end);
+        this.skills = "No Skills Entered";
+        this.status = "Application Pending";
+    }
+
+    //@@author Ridiculouswifi
+    /**
+     * Updates the status of the Internship with a valid status.
+     * Does not update with invalid statuses and will prompt for a valid status.
+     *
+     * @param userStatus user-inputted status.
+     */
+    public void updateStatus(String userStatus) throws InvalidStatus {
+        List<String> statuses = Arrays.asList("Application Pending", "Application Completed", "Accepted", "Rejected");
+        for (String status : statuses) {
+            if (status.equalsIgnoreCase(userStatus)) {
+                this.status = status;
+                return;
+            }
+        }
+        throw new InvalidStatus();
+    }
```

Comment:
You could consider using an enum here to make the code easier to update and expand on in the future.